### ISSUE TITLE
lasso-loader throws error for already loaded resources

### ISSUE THREAD

**Issue by schetnikovich** [lasso-js/lasso#175]
When using `require('lasso-loader').async(...)` for already loaded resources (JS, CSS etc), Lasso throws error similar to: `Loader metadata missing for "_0"`. 

The minimal page setup that reproduces this behavior is the following. 

Content of `browser.json`:

```json
{
  "dependencies": [
    "require-run: ./client.js"
  ]
}
```

Content of `client.js` (the only client script):

```js
window.onload = function() {
  require("lasso-loader").async(function(err) {
    var client = require('./client.js'); // we are loading "ourselves"
  });
};
```

The following error is thrown on page load: 

```
Loader metadata missing for "_0"
    at Object.async (http://localhost:8080/static/page.js:1183:15)
    at window.onload (http://localhost:8080/static/page.js:1212:62)
```

## But it works when you add not yet loaded resource

The same code will work when you add some not yet loaded resource. Here is `client.js` with single line added:

```js
window.onload = function() {
  require("lasso-loader").async(function(err) {
    var empty = require('./empty.js');    // ← File "empty.js" can be just empty
    var client = require('./client.js');
  });
};
```

In this case, no errors are thrown. 

<hr>

I have faced with this error in a context when two bundles load asynchronously each other. `Bundle1` successfully loads `Bundle2`, but then `Bundle2` fails to load `Bundle1`. It seems that the fact that `Bundle1` was already loaded, somehow prevents this operation to complete. Error message is the same as the one shown above. 

Let me know if I'm doing something wrong. Thank you!

Environment:

<table>
<tr><th>lasso</th><td>2.6.1</td></tr>
<tr><th>node</th><td>6.5.0</td></tr>
<tr><th>npm</th><td>3.10.3</td></tr>
<tr><th>OS</th><td>MacOS 10.12</td></tr>
</table>

**Comment by patrick-steele-idem** [comment]
Looks like a bug, but I can see why it is happening. It looks like `client.js` is already part of the initial page load and because of that there is nothing to asynchronously load. Because there is nothing to asynchronously load there is no writing of the metadata required for async loading. As a result of no metadata, you are seeing an error.

I'll look into a fix. Thank you for reporting the problem!

**Comment by schetnikovich** [comment]
Thank you, Patrick! Just a small comment to help you understand my "real" problem. Minimal page setup presented above is not a "real-world" problem. What i'm trying to do is the following:

`file1.js`:
```js
exports.load = function() {
  require("lasso-loader").async(function(err) {
    var client = require('./file2.js');
  });
};
```

`file2.js`:

```js
exports.load = function() {
  require("lasso-loader").async(function(err) {
    var client = require('./file1.js');
  });
};
```

Both `file1.js` and `file2.js` can be loaded asynchronously or as part of initial page load. Hope that this is possible with Lasso!

**Comment by philidem** [comment]
I fixed this issue and published `lasso@2.7.2`

**Comment by schetnikovich** [comment]
Just checked new `lasso@2.7.2` and it works as expected! Thank you, Phillip and Patrick!
